receptor_satellite/worker.py

```diff
@@ -163,13 +166,47 @@ def abort(self, error):
             host.mark_as_failed(error)
 
 
+async def cancel_run(satellite_api, run_id, queue, logger):
+    logger.info(f"Cancelling playbook run {run_id}")
+    run = await run_monitor.get(run_id)
+    status = None
+    if run is True:
+        logger.info(f"Playbook run {run_id} is already finished")
+        status = "finished"
+    elif run is None:
+        logger.info(f"Playbook run {run_id} is not known by receptor")
+        status = "failure"
+    else:
+        await satellite_api.init_session()
+        response = await satellite_api.cancel(run_id)
+        await satellite_api.close_session()
+        if response["status"] == 422:
+            status = "finished"
```

Comment:
Wouldn't you also want to mark this "Done" in the run_monitor?

Reply:
This would mean the job finished between checking with `run_monitor` and actually sending out the cancellation request to satellite so it should be marked as done by the thread which was handling the execution

To express the flow on a timeline:
1) Job is triggered by thread A
2) User wants to cancel the job
3) Thread B checks if the job is running
4) Job finishes, thread A marks it as done
5) Thread B calls to Satellite to cancel the job and gets back 422

Reply:
Okay - there's not a condition (even an error/failure condition) in which Satellite would finish the job but the polling never got to marking it done in memory?

Reply:
I just went over the code a few times. As long as the task in satellite finishes, it should get marked as done on receptor's side.

Reply:
I just went over the code a few times. As long as the task in satellite finishes, it should get marked as done on receptor's side.